app/models/chargeback_rate.rb

```diff
@@ -72,36 +74,67 @@ def self.seed_chargeback_rate_measure
     end
   end
 
+  def self.seed_cb_rate_currency
+    # seeding the chargeback_rate_detail_currencies
+    # Modified seed method. Now updates chargeback_rate_detail_currencies too
+    fixture_file_currency = File.join(FIXTURE_DIR, "chargeback_rate_detail_currencies.yml")
+    if File.exist?(fixture_file_currency)
+      fixture = YAML.load_file(fixture_file_currency)
+      fixture.each do |cbr|
+        rec = ChargebackRateDetailCurrency.find_by_name(cbr[:name])
+        if rec.nil?
+          _log.info("Creating [#{cbr[:name]}] with symbols=[#{cbr[:symbol]}]!!!!")
+          rec = ChargebackRateDetailCurrency.create(cbr)
+        else
+          fixture_mtime_currency = File.mtime(fixture_file_currency).utc
+          if fixture_mtime_currency > rec.created_at
+            _log.info("Updating [#{cbr[:name]}] with symbols=[#{cbr[:symbol]}]")
+            rec.update_attributes(cbr)
+            rec.created_at = fixture_mtime_currency
+            rec.save
+          end
+        end
+      end
+    end
+  end
+
   def self.seed_chargeback_rate
     # seeding the rates fixtures
     fixture_file = File.join(FIXTURE_DIR, "chargeback_rates.yml")
     fixture_file_measure = File.join(FIXTURE_DIR, "chargeback_rates_measures.yml")
+    fixture_file_currency = File.join(FIXTURE_DIR, "chargeback_rate_detail_currencies.yml")
 
     if File.exist?(fixture_file)
       fixture = YAML.load_file(fixture_file)
       fixture.each do |cbr|
         rec = find_by_guid(cbr[:guid])
         rates = cbr.delete(:rates)
+
         # The yml measure field is the name of the measure. It's changed to the id
         rates.each do |rate_detail|
           measure = ChargebackRateDetailMeasure.find_by(:name => rate_detail.delete(:measure))
+          currency = ChargebackRateDetailCurrency.find_by(:name => rate_detail.delete(:type_currency))
           unless measure.nil?
             rate_detail[:chargeback_rate_detail_measure_id] = measure.id
           end
+          unless currency.nil?
+            rate_detail[:chargeback_rate_detail_currency_id] = currency.id
+          end
         end
         if rec.nil?
           _log.info("Creating [#{cbr[:description]}] with guid=[#{cbr[:guid]}]")
           rec = create(cbr)
           rec.chargeback_rate_details.create(rates)
         else
-          fixture_mtime = File.mtime(fixture_file).utc
-          fixture_mtime_measure = File.mtime(fixture_file_measure).utc
-          if fixture_mtime > rec.created_on || fixture_mtime_measure > rec.created_on
+          fix_mtime = File.mtime(fixture_file).utc
+          fix_mtime_measure = File.mtime(fixture_file_measure).utc
+          fix_mtime_currency = File.mtime(fixture_file_currency).utc
```

Comment:
Same here, File.mtime is rather slow, we should do this once... I believe this is within the loop.